src/polygonParts/models/polygonPartsManager.ts

```diff
@@ -0,0 +1,166 @@
+      await this.createTables(ingestionContext);
+      await this.insert(ingestionContext);
+      await this.updatePolygonParts(ingestionContext);
+    });
+  }
+
+  private async createTables(ingestionContext: IngestionContext): Promise<void> {
+    const { entityManager, logger, polygonPartsPayload } = ingestionContext;
+
+    logger.debug(`creating polygon parts tables`);
+
+    try {
+      const createPolygonPartsProcedure = this.applicationConfig.createTablesStoredProcedure;
+      const entityName = this.getEntityName(polygonPartsPayload);
+      const entityQualifiedName = this.getDatabaseObjectQualifiedName(entityName);
+      await entityManager.query(`CALL ${createPolygonPartsProcedure}('${entityQualifiedName}');`);
+    } catch (error) {
+      const errorMessage = 'Could not create polygon parts tables';
+      logger.error({ msg: errorMessage, error });
+      throw new InternalServerError(errorMessage);
+    }
+  }
+
+  private async insert(ingestionContext: IngestionContext): Promise<void> {
+    const { entityManager, logger, polygonPartsPayload } = ingestionContext;
+    const { partsData, ...props } = polygonPartsPayload;
+
+    logger.debug(`inserting polygon parts data`);
+
+    // inserted props are ordered in the order of the columns of the entity, since the entity is not modeled directly by typeorm
+    const insertEntities: IngestionProperties[] = partsData.map((partData) => {
+      return {
+        productId: props.productId,
+        productType: props.productType,
+        catalogId: props.catalogId,
+        sourceId: partData.sourceId,
+        sourceName: partData.sourceName,
+        productVersion: props.productVersion,
+        ingestionDateUTC: undefined,
+        imagingTimeBeginUTC: partData.imagingTimeBeginUTC,
+        imagingTimeEndUTC: partData.imagingTimeEndUTC,
+        resolutionDegree: partData.resolutionDegree,
+        resolutionMeter: partData.resolutionMeter,
+        sourceResolutionMeter: partData.sourceResolutionMeter,
+        horizontalAccuracyCE90: partData.horizontalAccuracyCE90,
+        sensors: partData.sensors.join(this.applicationConfig.arraySeparator),
+        countries: partData.countries?.join(this.applicationConfig.arraySeparator),
+        cities: partData.cities?.join(this.applicationConfig.arraySeparator),
+        description: partData.description,
+        footprint: partData.footprint,
+      };
+    });
+    const entityName = this.getEntityName(polygonPartsPayload);
+    const entityQualifiedName = this.getDatabaseObjectQualifiedName(entityName);
+
+    try {
+      if (insertEntities.length === 1) {
+        // QueryBuilder API is used since insert() of a single record uses the object keys as fields
+        // which is unsuitable since the keys have a mapping to column names
+        await entityManager
+          .createQueryBuilder()
+          .insert()
+          .into<IngestionProperties>(`${entityQualifiedName}_parts`, [
```

Comment:
Try to find any solution for this hard-coded columns

Reply:
it's now derived